src/plugins/socketmon/private.h

```diff
@@ -220,19 +176,32 @@ struct tcp_endpoint_x64
     uint64_t owner;
 } __attribute__ ((packed));
 
+// Tested for Windows 8.1
+struct tcp_endpoint_win81_x64
+{
+    addr_t _pad1[2];      // +0x0
+    addr_t inetaf;        // +0x10 -> inetaf_win10_x64
+    addr_t addrinfo;      // +0x18
+    uint8_t _pad2[0x4c];  // +0x20
+    uint32_t state;       // +0x6c
+    uint16_t localport;   // +0x70
+    uint16_t remoteport;  // +0x72
+    uint8_t _pad3[0x1e4]; // +0x74
+    addr_t owner;         // +0x258
+} __attribute__((packed));
+
+// Tested for Windows 10 build 1803
 struct tcp_endpoint_win10_x64
 {
-    addr_t _pad1[2];
-    addr_t inetaf; // inetaf_win10_x64
-    addr_t addrinfo;
-    uint8_t _pad2[0x4c];
-    uint32_t state;
-    uint16_t localport;
-    uint16_t remoteport;
-    uint8_t _pad3[0x1E4];
-    addr_t owner;
-    addr_t _pad4;
-    addr_t createtime;
+    addr_t _pad1[2];      // +0x0
+    addr_t inetaf;        // +0x10 -> inetaf_win10_x64
+    addr_t addrinfo;      // +0x18
+    uint8_t _pad2[0x4c];  // +0x20
+    uint32_t state;       // +0x6c
+    uint16_t localport;   // +0x70
+    uint16_t remoteport;  // +0x72
+    uint8_t _pad3[0x204]; // +0x74
+    addr_t owner;         // +0x278
```

Comment:
It seems to be there are multiple versions now of this structure. You are dropping `createtime` and also change the location of where `owner` should be. This structure was created based on https://github.com/volatilityfoundation/volatility/blob/aa6b960c1077e447bda9d64df507ec02f8fcc958/volatility/plugins/overlays/windows/tcpip_vtypes.py#L270. So, if more recent versions of Windows 10 use the layout that you now specify, we should have a version check to decide which one to use.

Reply:
I don't know which Windows version the [line](https://github.com/volatilityfoundation/volatility/blob/aa6b960c1077e447bda9d64df507ec02f8fcc958/volatility/plugins/overlays/windows/tcpip_vtypes.py#L270) targets so I could not add version check. If you could specify the correct version for the structure you mentioned I would add version check.

Reply:
Well, you are saying the one you add works with 1803. The version that was specified worked with versions before that. So I would just go with that.

Reply:
@tklengyel, the only place where Windows build number is checked in `libvmi/libvmi/os/windows/core.c:find_windows_version_from_json_profile`. Should I patch LibVMI and export the function?

Reply:
Needs https://github.com/libvmi/libvmi/pull/868